app/src/main/java/com/imashnake/animite/data/sauce/NetworkBoundResource.kt

```diff
@@ -0,0 +1,16 @@
+package com.imashnake.animite.data.sauce
+
+import com.imashnake.animite.data.Resource
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+
+
+// TODO improve the handling of this, it's a bit too basic
+fun <T> Flow<T>.asState(): Flow<Resource<T?>> {
+    return map {
+        Resource.success<T?>(it)
```

Comment:
Need to incorporate a mapper function in here, and still not sure how I can emit Resource.error(message, ...) with a non-null object